### PR TITLE
i18n v2

### PR DESCRIPTION
**Pull request by leshakoss**

### CODE REVIEW

**Comment by albertorestifo** [comment]
Will this require changes/additions to the locale definitions?

**Comment by leshakoss** [comment]
@albertorestifo yeah, we will need help from I18n team.

We will need regexps for parse #219 function: https://github.com/date-fns/date-fns/pull/485/commits/950e46d4eb31378b8210d356d8884b82cb42c21c#diff-f041b0e48004ee1b9536bdfb7bec2602R6

long formats for this #390: https://github.com/date-fns/date-fns/pull/485/commits/950e46d4eb31378b8210d356d8884b82cb42c21c#diff-58ca79515d4154b95d179c47300ad43aR53

and formats for planned calendar date function #388: https://github.com/date-fns/date-fns/pull/485/commits/950e46d4eb31378b8210d356d8884b82cb42c21c#diff-58ca79515d4154b95d179c47300ad43aR53

And verify that everything is converted right, of course: https://github.com/date-fns/date-fns/blob/8fc8b1d507334ba8134a450da3c1401dd466f998/src/locale/it/_lib/buildLocalize/index.js#L5-L20

Locales API isn't exactly ready yet and will suffer from some reshuffling but in a few days I plan to release an alpha exactly for this reason. I will post a guide how to do v2 locales correctly soon. Stay tuned :-)

**Comment by edorivai** [comment]
Would be happy to help with `nl` translations.

File: src/locale/_lib/buildFormatLong/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+var tokensToBeShortedPattern = /MMMM|MM|DD|dddd/g
+
+function buildShortLongFormat (format) {
+  return format.replace(tokensToBeShortedPattern, function (token) {
+    return token.slice(1)
+  })
+}
+
+export default function buildFormatLong (obj) {
```
Comment (kossnocorp) [inline src/locale/_lib/buildFormatLong/index.js]: Please add JSDoc to locale helpers

File: src/locale/en-US/_lib/match/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+var matchMonthsPatterns = {
+  short: /^(jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec)/i,
+  long: /^(january|february|march|april|may|june|july|august|september|october|november|december)/i
+}
+
+var parseMonthPatterns = {
+  any: [/^ja/i, /^f/i, /^mar/i, /^ap/i, /^may/i, /^jun/i, /^jul/i, /^au/i, /^s/i, /^o/i, /^n/i, /^d/i]
+}
+
```
Comment (kossnocorp) [inline src/locale/en-US/_lib/match/index.js]: Please add a note what to do when a locale doesn't have am/pm

File: src/locale/ru/_lib/localize/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+function ordinalNumber (dirtyNumber, dirtyOptions) {
+  var options = dirtyOptions || {}
+  var unit = String(options.unit)
+  var suffix
+
+  if (unit === 'dayOfMonth') {
+    suffix = '-е'
+  } else if (unit === 'isoWeek') {
```
Comment (kossnocorp) [inline src/locale/ru/_lib/localize/index.js]: `|| unit === 'week'`

File: src/locale/ru/_lib/match/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  short: /^(вск|вос|пнд|пон|втр|вто|срд|сре|чтв|чет|птн|пят|суб).?/i,
+  long: /^(воскресень[ея]|понедельника?|вторника?|сред[аы]|четверга?|пятниц[аы]|суббот[аы])/i
+}
+
+var parseWeekdayPatterns = {
+  any: [/^в[ос]/i, /^п[он]/i, /^в/i, /^ср/i, /^ч/i, /^п[ят]/i, /^с[уб]/i]
+}
+
+var matchMonthsPatterns = {
+  short: /^(янв|фев|март?|апр|ма[йя]|июн[ья]?|июл[ья]?|авг|сент?|окт|нояб?|дек).?/i,
```
Comment (kossnocorp) [inline src/locale/ru/_lib/match/index.js]: `).?` -> `)\.?)`

**Comment by leshakoss** [comment]
v2 locale creator guide coming soon 😃

**Pull request merged**